Makefile

```diff
@@ -24,7 +24,8 @@ clean:
 
 .PHONY: test
 test:
-	go test ./... -count=1 -race
+	mkdir -p $(TEST_COVERAGE_DIR)
+	go test ./... -count=1 -race -coverprofile=$(TEST_COVERAGE_DIR)/coverage.out -covermode=atomic
```

Comment:
Use `-covermode=count` instead; `atomic` is more precise for parallel algorithms, but [it is expensive](https://go.dev/blog/cover) and we do not need a perfect heat map